tests/dummy/app/states/todos.js

```diff
@@ -1,6 +1,8 @@
 import BufferedProxy from 'ember-buffered-proxy/proxy';
 
-BufferedProxy.reopenClass({
+const TodosProxy = BufferedProxy.extend();
+
+TodosProxy.reopenClass({
   initialState() {
     return {
       content: this.get('todo')
```

Comment:
I'm realizing that perhaps I don't understand `reopenClass` like I thought I did. I thought that `reopenClass` would affect **all** `BufferedProxy` packages, not just the one that I'm exporting. I was trying to make an example app that would show this, but it's not working like I expected (it's working how you're describing it). But maybe you can explain to me why I'm confused.

In [this example](https://github.com/blimmer/ember-cli-example-app-for-github/tree/ember-state-services-44) I create two state services, [foo.js](https://github.com/blimmer/ember-cli-example-app-for-github/blob/ember-state-services-44/app/states/foo.js) and [bar.js](https://github.com/blimmer/ember-cli-example-app-for-github/blob/ember-state-services-44/app/states/bar.js).

In foo.js, I create an `initialState` that [gets model1](https://github.com/blimmer/ember-cli-example-app-for-github/blob/ember-state-services-44/app/states/foo.js#L10).

In bar.js, I create an `initialState` that [gets model2](https://github.com/blimmer/ember-cli-example-app-for-github/blob/ember-state-services-44/app/states/bar.js#L10).

When debugging in the state services, I can see that `BufferedProxy.initialState` is undefined first, and then defined the next time it's reopened (because we reopened the class).

However, when I place the corresponding components, I'm seeing them initialize properly. Can you guys help me understand why this works? I was under the impression that the "last re-open that defines `initialState` would override different logic in any other re-open".